### PR TITLE
v2 pipeline ETH task types

### PR DESCRIPTION
**Pull request by spooktheducks**
- Added ETH/CBOR task types:
    - `ethcall`
    - `ethtx`
    - `ethabiencode`
    - `ethabidecode`
    - `ethabidecodelog`
    - `cborparse`
- Removed `pipelineInput` param, replaced with `$(vars)`
    - All job types now inject `$(jobSpec.databaseID / externalJobID / name)` into `vars` before running
    - All job types now inject `$(jobRun.meta)` (although current `meta` behavior is unchanged) as well as job type-dependent variables
    - Changed `$(input)` to `$(jobRun.<job type-dependent field>)`
- Simplification/standardization: changed `log.Broadcaster#Register` options to only accept `LogsWithTopics`

[ch10959]

### CODE REVIEW

File: core/services/pipeline/common_eth.go [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+			argName  []byte
+			indexed  bool
+		)
+		for i := range parts {
+			parts[i] = bytes.TrimSpace(parts[i])
+			if len(parts[i]) > 0 {
+				argParts = append(argParts, parts[i])
+			}
+		}
+		switch len(argParts) {
```
Comment (archseer) [inline core/services/pipeline/common_eth.go:46]: All the handwritten parsing here seems fine but I wonder if we could construct an actual parser for this.

Comment (archseer) [reply core/services/pipeline/common_eth.go:46]: Note: can't find a decent parser combinator library for Go so it might be too much of a hassle

**Review by archseer — COMMENTED** [review]
Seemed okay, I mainly looked at the `services/pipeline/` bits.

File: core/services/directrequest/delegate.go [hunk 2]
```diff
@@ -279,7 +280,25 @@ func (l *listener) handleOracleRequest(request *oracle_wrapper.OracleOracleReque
...
+				"logBlockHash":   request.Raw.BlockHash,
+				"logBlockNumber": request.Raw.BlockNumber,
+				"logTxHash":      request.Raw.TxHash,
+				"logAddress":     request.Raw.Address,
+				"logTopics":      request.Raw.Topics,
+				"logData":        request.Raw.Data,
+			},
+		})
+
+		run, trrs, err := l.pipelineRunner.ExecuteRun(ctx, *l.job.PipelineSpec, vars, pipeline.JSONSerializable{Val: meta, Null: false}, *logger)
```
Comment (connorwstein) [inline core/services/directrequest/delegate.go:301]: oh do we even need the separate meta argument here if its available in the vars?

File: core/services/pipeline/common_eth.go (hunk 1 above)
Comment (spooktheducks) [reply core/services/pipeline/common_eth.go:46]: Yeah the one I'm familiar with seemed like overkill for this

File: core/services/offchainreporting/contract_tracker.go [hunk 3]
```diff
@@ -31,6 +30,11 @@ import (
 	ocrtypes "github.com/smartcontractkit/libocr/offchainreporting/types"
 )
 
+// configMailboxSanityLimit is the maximum number of configs that can be held
```
Comment (connorwstein) [inline core/services/offchainreporting/contract_tracker.go:33]: double comment?

File: core/services/pipeline/runner.go [hunk 4]
```diff
@@ -8,49 +8,53 @@ import (
...
+	"github.com/smartcontractkit/chainlink/core/service"
+	"github.com/smartcontractkit/chainlink/core/services/eth"
+	"github.com/smartcontractkit/chainlink/core/store/models"
 )
 
 //go:generate mockery --name Runner --output ./mocks/ --case=underscore
 
-type Runner interface {
-	service.Service
+type (
```
Comment (connorwstein) [inline core/services/pipeline/runner.go:27]: nit: didn't we get to the bottom of this earlier in that doing the big type def is not idiomatic go?

File: core/services/pipeline/task.cborparse.go [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	"go.uber.org/multierr"
+
+	"github.com/smartcontractkit/chainlink/core/store/models"
+)
+
+type CBORParseTask struct {
```
Comment (connorwstein) [inline core/services/pipeline/task.cborparse.go:12]: sorry what uses this again? Is it that some people stuff json objects into a solidity ```bytes``` parameter?

File: core/services/pipeline/common_eth.go (hunk 1 above)
Comment (connorwstein) [reply core/services/pipeline/common_eth.go:46]: We could just put the contract abi json in the spec and then use abigen tools directly to parse it? We already have json elsewhere in our specs and the spec author would already have the contract abi json available

File: core/services/pipeline/task.eth_tx.go [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+		errors.Wrap(ResolveParam(&toAddr, From(NonemptyString(t.To))), "to"),
+		errors.Wrap(ResolveParam(&data, From(VarExpr(t.Data, vars), JSONWithVarExprs(t.Data, vars, false), nil)), "data"),
+		errors.Wrap(ResolveParam(&gasLimit, From(VarExpr(t.GasLimit, vars), NonemptyString(t.GasLimit), t.config.EthGasLimitDefault())), "gasLimit"),
+		errors.Wrap(ResolveParam(&txMeta, From(VarExpr(t.TxMeta, vars), JSONWithVarExprs(t.TxMeta, vars, false), MapParam{})), "txMeta"),
+	)
+	if err != nil {
+		return Result{Error: err}
+	}
+
+	fromAddr, err := t.keyStore.GetRoundRobinAddress(fromAddrs...)
```
Comment (connorwstein) [inline core/services/pipeline/task.eth_tx.go:67]: we want ```GetRoundRobinAddress``` and ```CreateEthTransaction``` to be in a pg tx together

File: core/services/directrequest/delegate.go (hunk 2 above)
Comment (spooktheducks) [reply core/services/directrequest/delegate.go:301]: Nope we can just use vars from here forward, but I wanted to save that small refactor for a separate PR

File: core/services/pipeline/task.cborparse.go (hunk 5 above)
Comment (spooktheducks) [reply core/services/pipeline/task.cborparse.go:12]: Yeah, in direct request jobs the `bytes data` field is a CBOR-encoded map

File: core/services/pipeline/common_eth.go (hunk 1 above)
Comment (spooktheducks) [reply core/services/pipeline/common_eth.go:46]: We're going to reach out to DevRel to do a survey of devs about this, will move to a subsequent PR if a change is needed

File: core/services/directrequest/delegate.go (hunk 2 above)
Comment (archseer) [reply core/services/directrequest/delegate.go:301]: I'll handle this in `pipeline-async-jobs`, it simplifies some code for me there

File: core/services/pipeline/task.eth_tx.go (hunk 6 above)
Comment (connorwstein) [reply core/services/pipeline/task.eth_tx.go:101]: spoke offline GetRoundRobinAddress doesnt touch the db so its fine

**Review by connorwstein — APPROVED** [review]

**Pull request merged**